src/rgb/schema/types.rs

```diff
@@ -14,12 +14,22 @@
 
 use std::{io, convert::TryFrom};
 
-use num_integer::Integer;
 use num_traits::{ToPrimitive, FromPrimitive};
 use num_derive::{ToPrimitive, FromPrimitive};
 
 use crate::csv::serialize::*;
 
+pub trait UnsignedInteger: Clone + Copy + PartialEq + Eq + PartialOrd + Ord + Into<u64> + std::fmt::Debug {
```

Comment:
I assume we do not need this as pub? Otherwise it's better to move it into `common` mod.